### PR TITLE
feat: add SideEffect type placeholder for scalasdk

### PR DESCRIPTION
**Pull request by octonato**
not full impl, just so we can use this type when defining the new components Effects

### CODE REVIEW

File: sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/SideEffect.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   * Create a side effect of the given service call.
+   *
+   * @param serviceCall
+   *   The service call to effect.
+   * @param synchronous
+   *   Whether this effect should be executed synchronously.
+   * @return
+   *   The side effect.
+   */
+  def of(serviceCall: ServiceCall, synchronous: Boolean): SideEffect = ???
```
Comment (patriknw) [inline sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/SideEffect.scala:32]: Is there a reason for not naming it `apply`?

Comment (octonato) [reply sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/SideEffect.scala:32]: yes, me doing mechanical copy-and-paste. 

I will change

Comment (octonato) [reply sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/SideEffect.scala]: done

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Pull request merged**